### PR TITLE
refactor: drop jshandle factory from ExecutionContext

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch:
- merges `ElementHandle` into `ExecutionContext` (for simplicity; there's no good reason to have them in separate files).
- removes the necessity to pass handle factory when creating `ExecutionContext`

This makes it easier to create execution contexts out of payloads.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "node": ">=6.4.0"
   },
   "puppeteer": {
-    "chromium_revision": "579032"
```
Comment (JoelEinbinder) [inline package.json]: what

File: lib/Worker.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ class Worker extends EventEmitter {
     let jsHandleFactory;
     this._client.once('Runtime.executionContextCreated', async event => {
       jsHandleFactory = remoteObject => new JSHandle(executionContext, client, remoteObject);
```
Comment (JoelEinbinder) [inline lib/Worker.js:35]: remove jsHandleFactory

File: lib/Page.js [hunk 3]
```diff
@@ -27,6 +27,7 @@ const Tracing = require('./Tracing');
 const {helper, debugError, assert} = require('./helper');
 const {Coverage} = require('./Coverage');
 const {Worker} = require('./Worker');
+const {createJSHandle} = require('./ExecutionContext');
```
Comment (JoelEinbinder) [inline lib/Page.js:30]: JSHandleFromProtocol

File: lib/FrameManager.js [hunk 4]
```diff
@@ -259,17 +254,19 @@ FrameManager.Events = {
...
 class Frame {
   /**
+   * @param {!FrameManager} frameManager
    * @param {!Puppeteer.CDPSession} client
    * @param {?Frame} parentFrame
    * @param {string} frameId
    */
-  constructor(client, parentFrame, frameId) {
+  constructor(frameManager, client, parentFrame, frameId) {
+    this._frameManager = frameManager;
```
Comment (JoelEinbinder) [inline lib/FrameManager.js:263]: page

**Review by JoelEinbinder — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (aslushnikov) [reply package.json]: rebaselined.

File: lib/FrameManager.js (hunk 4 above)
Comment (aslushnikov) [reply lib/FrameManager.js:263]: done

Comment (aslushnikov) [reply lib/FrameManager.js:263]: Not done; we need framemanager in element handle to resolve content frames

**Pull request merged**